### DISCUSSION TITLE
Floyd-Warshall algortithms in dense.py don't check for negative cycles

### DISCUSSION THREAD

**Discussion by paulitapb** [networkx/networkx#6125] [Algorithms]
I was looking at Floyd-Warshall implementations and they don't check for negative cycles. The doc says that FW doesn't work with graphs with negative cycles but I think It'll be useful to raise an exception and print a message in that case. I was thinking of adding this but I'm interested in some opinions. Is there some reason this isn't checked?

**Comment by dschult** [comment]
What *does* happen when you provide a network with a negative cycle in it?

**Reply by paulitapb** [reply]
Well, the minimum weighted path is not properly defined because you can always go through the cycle and have a new minimum. The current implementation will return a dictionary with the weights of the minimum paths as they were the optimal solution but that's not true. It gives you the weight of the minimum path with a bounded length because you can go through the cycle a finite amount of times. Given that is very simple to check if there are negative cycles within the algorithm I think it will be useful to return a warning in that case.
